### ISSUE TITLE
Add new runtime tests to validate the order of probe firing

### ISSUE THREAD

**Issue by ajor** [bpftrace/bpftrace#3170]
Multiple probes are allowed to be attached to the same event. When this happens, bpftrace tries to ensure that the probes fire in the order that they're defined in the script.

e.g. this script should always print a sequence of "abc" and not "cba":
```
tracepoint:syscalls:sys_enter_read { printf("a") }
tracepoint:syscalls:sys_enter_read { printf("b") }
tracepoint:syscalls:sys_enter_read { printf("c") }
```

For reference, this is the code that controls the order in which we load duplicate probes and defines the order they fire in: https://github.com/bpftrace/bpftrace/blob/bc73244963f206814ae45ec78ebe52cd389f6381/src/bpftrace.cpp#L952-L976

This area of code may be changing in the future, with #3128 and/or #2334, so it'd be good to add some test coverage.

For as many of the different probe types as is practical, we should try to write new runtime tests that confirm they fire in the correct order.

**Comment by shawnh2** [comment]
I'd like to give it a try, can you assign me?

**Comment by viktormalik** [comment]
@shawnh2 sure, let us know if you need any help. Thanks!

**Comment by shawnh2** [comment]
Hi, sorry for the delay of this issue.

This runtime test, do you mean like: https://github.com/bpftrace/bpftrace/blob/1517d1cc9b7436069412d764ae32d34c9e58afd7/tests/runtime/probe#L352-L355

and

https://github.com/bpftrace/bpftrace/blob/master/tests/runtime/scripts/tracepoint_order.bt 

but for other probes ?

**Comment by jordalgo** [comment]
Maybe another test for probes that are attached in reverse order would be good.

**Comment by ajor** [comment]
Yes, exactly like the tracepoint_order/kprobe_order/etc. tests (I'd forgotten we had those!), but for all the probe types we support.
